src/lib.rs

```diff
@@ -281,29 +339,21 @@ impl Infer {
     /// info.add("custom/foo", "foo", custom_matcher);
     /// let v = vec![0x10, 0x11, 0x12, 0x13];
     /// let res =  info.get(&v).unwrap();
-    /// assert_eq!("custom/foo", res.mime);
-    /// assert_eq!("foo", res.ext);
+    /// assert_eq!("custom/foo", res.mime_type());
+    /// assert_eq!("foo", res.extension());
     /// ```
-    pub fn add(&mut self, mime: &str, ext: &str, m: Matcher) {
-        self.mmap.push((
-            map::MatcherType::CUSTOM,
-            mime.to_string(),
-            ext.to_string(),
+    pub fn add(&mut self, mime_type: &'static str, extension: &'static str, m: Matcher) {
```

Comment:
One notable change I had to make is require these strings to be static. I don't think this will be a problem, since users declaring custom matchers will use static strings to do so, but still think it's useful to highlight this change.

Reply:
I believe this is reasonable indeed.

There might be situations where one would need none static str : If you want to parse mime types from IANA or other static DB for instance. In such case I would expect the user to contribute back here instead of building an entire DB on its own.

We will see if users raise issues about that. Anyway if we consider `no_alloc` an important feature (I believe it is) then we have no choice and this is a compromise to accept.

Reply:
>  If you want to parse mime types from IANA or other static DB for instance. In such case I would expect the user to contribute back here instead of building an entire DB on its own.

I guess they would have to do it at compile time, since they also have to pass in a matcher function. It's not that important to have this also be `no_alloc`, I just feel a crate like this should be able to depend just on `core`.